pyo3-macros-backend/src/method.rs

```diff
@@ -69,24 +69,6 @@ fn handle_argument_error(pat: &syn::Pat) -> syn::Error {
     syn::Error::new(span, msg)
 }
 
-#[derive(Clone, PartialEq, Debug, Copy, Eq)]
-pub enum MethodTypeAttribute {
-    /// `#[new]`
-    New,
-    /// `#[new]` && `#[classmethod]`
-    NewClassMethod,
```

Comment:
What is the thought behind yeeting the `NewClassMethod` variant? I have no particular opinion - just curious.

Reply:
A lot of the complexity from the old implementation came from the need to work out when this variant was being built out of the two attributes. By removing it and instead pattern matching this combination I think the code reads better.